validator/sawtooth_validator/gossip/identity_observer.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import logging
+
+from sawtooth_validator.journal.chain import ChainObserver
+from sawtooth_validator.journal.event_extractors import \
+    BlockEventExtractor
+from sawtooth_validator.journal.event_extractors import \
+    ReceiptEventExtractor
+from sawtooth_validator.server.events.subscription import EventSubscription
+
+LOGGER = logging.getLogger(__name__)
+
+
+class IdentityObserver(ChainObserver):
+    """
+    The Identity Observer is used to update the local permission verifier's
+    caches of identity data when an Identity transaction is committed.
+    """
+    def __init__(self, to_update, forked):
+        # function to notify the permission verifier to update the caches
+        self.to_update = to_update
+        # function to notify the permission verifier there has been a
+        # fork and that the entries in the cache should be invalidated
+        self.forked = forked
+        self.last_block_id = None
+
+    def chain_update(self, block, receipts):
+        """
+        Handles both "block_commit" Events and "identity_update" Events. For
+        "block_commit", the last_block_num is updated or a fork is detected.
+        For "identity_update", the corresponding cache entry will be updated.
+        """
+
+        block_events = BlockEventExtractor(block).extract([
+            EventSubscription(event_type="block_commit")])
+        receipt_events = ReceiptEventExtractor(receipts).extract([
+            EventSubscription(event_type="identity_update")])
+
+        for event in block_events:
+            if event.event_type == "block_commit":
```

Comment:
This check is unnecessary because you are only subscribed to block_commit events (assuming the extractor works correctly..).